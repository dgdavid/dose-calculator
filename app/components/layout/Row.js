import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';

const Row = ({ size, style, children, ...props }) => {
  const styles = [
    style,
    { flex: size, flexDirection: 'row' },
  ];

  return (
    <View
      style={styles}
      {...props}
    >
      { children }
    </View>
  );
};

Row.propTypes = {
  size: PropTypes.number.isRequired,
  style: ViewPropTypes.style,
  children: PropTypes.node,
};

Row.defaultProps = {
  size: 1,
};

export default Row;
