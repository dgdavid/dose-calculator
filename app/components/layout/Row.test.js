import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Row from './Row';

describe('render correctly', () => {
  test('receiving the `size` prop', () => {
    const tree = renderer.create(<Row size={2} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('without `size` prop', () => {
    const tree = renderer.create(<Row size={2} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
