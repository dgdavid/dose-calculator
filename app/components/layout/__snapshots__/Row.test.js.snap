// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly receiving the \`size\` prop 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flex": 2,
        "flexDirection": "row",
      },
    ]
  }
/>
`;

exports[`render correctly without \`size\` prop 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flex": 2,
        "flexDirection": "row",
      },
    ]
  }
/>
`;
